lib/Runtime/Debug/TTSnapObjects.cpp

```diff
@@ -2432,21 +2474,128 @@ namespace TTD
+
+        void DoAddtlValueInstantiation_SnapJavascriptPromiseAsyncSpawnExecutorFunction(const SnapObject* snpObject, Js::RecyclableObject* obj, InflateMap* inflator)
         {
-            // TODO
-            return nullptr;
         }
 
-        void DoAddtlValueInstantiation_SnapAsyncFunction(const SnapObject* snpObject, Js::RecyclableObject* obj, InflateMap* inflator) {}
-        void EmitAddtlInfo_SnapAsyncFunction(const SnapObject* snpObject, FileWriter* writer) {}
-        void ParseAddtlInfo_SnapAsyncFunction(SnapObject* snpObject, FileReader* reader, SlabAllocator& alloc) {}
+        void EmitAddtlInfo_SnapJavascriptPromiseAsyncSpawnExecutorFunction(const SnapObject* snpObject, FileWriter* writer)
+        {
+            SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo* info = SnapObjectGetAddtlInfoAs<SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo*, SnapObjectType::JavascriptPromiseAsyncSpawnExecutorFunction>(snpObject);
+            writer->WriteAddr(NSTokens::Key::objectId, info->generator, NSTokens::Separator::CommaSeparator);
+            writer->WriteKey(NSTokens::Key::target, NSTokens::Separator::CommaSeparator);
+            NSSnapValues::EmitTTDVar(info->target, writer, NSTokens::Separator::NoSeparator);
+        }
+
+        void ParseAddtlInfo_SnapJavascriptPromiseAsyncSpawnExecutorFunction(SnapObject* snpObject, FileReader* reader, SlabAllocator& alloc)
+        {
+            SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo* info = alloc.SlabAllocateStruct<SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo>();
+            info->generator = reader->ReadAddr(NSTokens::Key::objectId, true);
+            reader->ReadKey(NSTokens::Key::target, true);
+            info->target= NSSnapValues::ParseTTDVar(false, reader);
+            SnapObjectSetAddtlInfoAs<SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo*, SnapObjectType::JavascriptPromiseAsyncSpawnExecutorFunction>(snpObject, info);
+        }
 #if ENABLE_SNAPSHOT_COMPARE
-        void AssertSnapEquiv_SnapAsyncFunction(const SnapObject* sobj1, const SnapObject* sobj2, TTDCompareMap& compareMap)
+        void AssertSnapEquiv_SnapJavascriptPromiseAsyncSpawnExecutorFunction(const SnapObject* sobj1, const SnapObject* sobj2, TTDCompareMap& compareMap)
         {
         }
 #endif
 
+        ////////////////////
+
+        Js::RecyclableObject *DoObjectInflation_SnapJavascriptPromiseAsyncSpawnStepArgumentExecutorFunctionInfo(const SnapObject *snpObject, InflateMap *inflator)
+        {
+            Js::ScriptContext *ctx = inflator->LookupScriptContext(snpObject->SnapType->ScriptContextLogId);
+            SnapJavascriptPromiseAsyncSpawnStepArgumentExecutorFunctionInfo* info = SnapObjectGetAddtlInfoAs<SnapJavascriptPromiseAsyncSpawnStepArgumentExecutorFunctionInfo *, SnapObjectType::JavascriptPromiseAsyncSpawnStepArgumentExecutorFunction>(snpObject);
+            Field(Js::Var) reject = (info->reject != nullptr) ? inflator->InflateTTDVar(info->reject) : nullptr;
```

Comment:
Are these `Field` annotations necessary? My understanding is that they are needed if we are going to write to recycler allocated memory, but these all seem to just be local references to existing objects?

Reply:
I don't have a really good grasp of this. :(.  I know there were some recent fixes around usage here, but I'm not clear on when we need the `Field` macros & when we don't.  I was using these here to just be consistent w/ what is defined in the `JavascriptGeneratorFunction`  class.  I left them in around the `Js::Var` items, but removed it around the `Js::JavascriptGenerator*`, since `Field` is not being used in other places for Js::* pointers.

Reply:
I am pretty sure you don't need them here since these pointers are on the stack so the GC is already aware of them.

Reply:
I removed.